source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,172 @@
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/simulator.h>
+#include <aspect/utilities.h>
+#include <aspect/postprocess/plateness_statistics.h>
+#include <aspect/global.h>
+#include <aspect/geometry_model/spherical_shell.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    PlatenessStatistics<dim>::execute (TableHandler &statistics)
+    {
+      const types::boundary_id top_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+
+      // create a quadrature formula for the velocity.
+      const QGauss<dim-1> quadrature_formula (this->introspection().polynomial_degree.velocities+1);
+   
+      FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                        this->get_fe(),
+                                        quadrature_formula,
+                                        update_gradients |
+                                        update_quadrature_points |
+                                        update_JxW_values);
+         
+      // Vector to store the integral of the second invariant of the strain rate
+      // over the surface.
+      double local_second_invariant_of_strain_rate_integral = 0.0;
+      double local_surface_area_integral = 0.0;
+
+      std::vector<std::pair<double,double>> local_second_invariant_of_strain_rate_and_corresponding_area; 
+      std::vector<SymmetricTensor<2,dim>> strain_rate (quadrature_formula.size());
+ 
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned() && cell->at_boundary())
+          {
+            unsigned int face_idx = numbers::invalid_unsigned_int;
+            bool at_upper_surface = false;
+            {
+              for (unsigned int f=0; f<cell->n_faces(); ++f)
+                {
+                  if (cell->at_boundary(f) && (cell->face(f)->boundary_id() == top_boundary_id))
+                    {
+                      // If the cell is at the top boundary, assign face_idx.
+                      face_idx = f;
+                      at_upper_surface = true;
+                      break;
+                    }
+                }
+
+            }
+            if (at_upper_surface)
+              {
+                fe_face_values.reinit(cell, face_idx);
+                fe_face_values[this->introspection().extractors.velocities].get_function_symmetric_gradients (this->get_solution(), strain_rate);
+              
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    local_second_invariant_of_strain_rate_and_corresponding_area[q].first = std::max(std::sqrt(std::fabs(second_invariant(deviator(strain_rate[q])))),min_strain_rate);
```

Comment:
This will probably compile but not run. Note how you are not setting the size of  `local_second_invariant_of_strain_rate_and_corresponding_area` at the top, so it's an empty array. Indexing the `[q]`'th element is likely going to give you a segmentation fault.